compiler/rustc_trait_selection/src/solve/fulfill.rs

```diff
@@ -48,29 +51,46 @@ impl<'tcx> TraitEngine<'tcx> for FulfillmentCtxt<'tcx> {
         if self.obligations.is_empty() {
             Vec::new()
         } else {
-            unimplemented!("ambiguous obligations")
+            self.obligations
+                .drain(..)
+                .map(|obligation| FulfillmentError {
+                    obligation: obligation.clone(),
+                    code: FulfillmentErrorCode::CodeSelectionError(SelectionError::Unimplemented),
+                    root_obligation: obligation,
+                })
+                .collect()
         }
```

Comment:
I guess this `if` condition can now be removed and we can just use the else path?